assets/prototype/domain/eventEditor/containers/mutations/useEntityMutation.js

```diff
@@ -0,0 +1,146 @@
+import { useApolloClient } from '@apollo/react-hooks';
+import { ApolloError } from 'apollo-client';
+import { useState } from '@wordpress/element';
+
+import { mutations } from './';
+import useMutators from './useMutators';
+
+/**
+ * @param {string} type Entity type name
+ */
+const useEntityMutation = (type, id = '') => {
+	const client = useApolloClient();
+	const [result, setResult] = useState({ called: false, loading: false });
+	const mutators = useMutators();
+
+	/**
+	 * @param {string} mutation Type of mutation - CREATE|UPDATE|DELETE
+	 */
+	const getMutation = (mutationType = 'UPDATE') => {
+		// For example "CREATE_DATETIME"
+		const mutaton = `${mutationType}_${type.toUpperCase()}`;
+		return mutations[mutaton];
+	};
+
+	/**
+	 * @param {string} mutation Type of mutation
+	 * @param {object} data     Mutation data
+	 */
+	const getMutationOptions = (mutationType, data = {}) => {
+		// e.g. "datetmeMutator"
+		const key = `${type.toLowerCase()}Mutator`;
+		const { [key]: mutator } = mutators;
```

Comment:
You lost me here :confused:

Reply:
See `useMutators` hook, you will understand.

Reply:
dynamically destructuring from an object using `key` and then renaming that var to `mutator` right?

Reply:
Yup!